web/tests/model.spec.ts

```diff
@@ -0,0 +1,64 @@
+// Tests machine learning boxes.
+import { test, expect } from '@playwright/test';
+import { Workspace } from './lynxkite';
+
+test('ML with Spark', async ({ page }) => {
+  const ws = await Workspace.empty(page, 'model');
+  // Create a graph.
+  await ws.addBox({
+    id: 'create', name: 'Create graph in Python', x: 100, y: 100, params: {
+      outputs: 'vs.age: float, vs.yob: float',
+      code: `
+    vs = pd.DataFrame({"age": [25, 40, 60, 35, 50], "yob": [1990, 1975, 1955, 1980, 1965]})
+    `.trim(),
+    }
+  });
+  let state = await ws.openStateView('create', 'graph');
+  await expect(state.left.vertexCount).toHaveText('5');
+  await state.close();
+
+  // Train a model.
+  await ws.addBox({
+    id: 'train', name: 'Train linear regression model',
+    x: 100, y: 200, after: 'create', params: {
+      name: 'age_from_yob',
+      label: 'age',
+      features: ['yob'],
+    },
+  });
+  state = await ws.openStateView('train', 'graph');
+  const model = state.left.graphAttribute('age_from_yob');
+  await expect(model.value).toHaveText('Linear regression model predicting age');
+  const p = await model.popup();
+  await expect(p.locator('#model-method')).toHaveText('Linear regression');
+  await expect(p.locator('#model-label')).toHaveText('age');
+  await expect(p.locator('#model-features')).toHaveText('yob');
+  await expect(p.locator('#model-details:visible')).toHaveText(/intercept\s*2015/);
```

Comment:
We could just have different IDs for different elements.

Reply:
> We could just have different IDs for different elements.

Sorry, don't know why I just swept it under the rug! I think when I saw `ng-show` I thought it was two elements that are semantically one, and we just display one or the other. But that's not the case! The second element has the same ID because it was copy-pasted and we didn't notice the ID!

Thanks. Fixed now.